Code/Framework/AzCore/AzCore/Settings/SettingsRegistryMergeUtils.cpp

```diff
@@ -796,20 +796,27 @@ namespace AZ::SettingsRegistryMergeUtils
 
     void MergeSettingsToRegistry_ManifestGemsPaths(SettingsRegistryInterface& registry)
     {
-        auto MergeGemPathToRegistry = [&registry](AZStd::string_view manifestKey,
-            AZStd::string_view gemName,
-            AZ::IO::PathView gemRootPath)
+        // cache a vector so that we don't mutate the registry while inside visitor iteration.
+        AZStd::vector<AZStd::pair<AZStd::string, AZ::IO::FixedMaxPath>> collectedGems;
+        auto CollectManifestGems =
+            [&collectedGems](AZStd::string_view manifestKey, AZStd::string_view gemName, AZ::IO::PathView gemRootPath)
         {
-            using FixedValueString = SettingsRegistryInterface::FixedValueString;
             if (manifestKey == GemNameKey)
             {
-                const auto manifestGemJsonPath = FixedValueString::format("%s/%.*s/Path",
-                    ManifestGemsRootKey, AZ_STRING_ARG(gemName));
-                registry.Set(manifestGemJsonPath, gemRootPath.LexicallyNormal().Native());
+                collectedGems.push_back(AZStd::make_pair(AZStd::string(gemName), AZ::IO::FixedMaxPath(gemRootPath)));
             }
         };
 
-        VisitAllManifestGems(registry, MergeGemPathToRegistry);
+        VisitAllManifestGems(registry, CollectManifestGems);
+
+        for (const auto& element : collectedGems)
+        {
+            const AZStd::string& gemName = element.first;
+            const AZ::IO::FixedMaxPath& gemRootPath = element.second;
```

Comment:
Nit: I believe you could write this more concisely as
```
for (const auto& [gemName, gemRootPath] : collectedGems)
```

Reply:
I'll update this.  I did one update that fixed the semicolon but I'll add a comment to the other one and use the for loop change.